### PR TITLE
Nodejs: rename block types

### LINKED ISSUE #1635: nodejs: rename block types
https://github.com/iotaledger/iota-sdk/issues/1515

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Rename block types to match Rust https://github.com/iotaledger/iota-sdk/pull/1626

## Links to any relevant issues

Closes #1635 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

`tsc`

### CODE REVIEW

File: bindings/nodejs/lib/types/block/core/block.ts [hunk 1]
```diff
@@ -1,67 +1,222 @@
...
-     * NOTE: Will throw an error if the block is not a `ValidationBlock`.
+     * Gets the block body as an actual `ValidationBlockBody`.
+     * NOTE: Will throw an error if the block body is not a `ValidationBlockBody`.
      * @returns The block
      */
-    asValidation(): ValidationBlock {
+    asValidation(): ValidationBlockBody {
         if (this.isValidation()) {
-            return this as unknown as ValidationBlock;
+            return this.body as unknown as ValidationBlockBody;
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/types/block/core/block.ts]: Can we just call the inner methods?

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/block/core/block.ts]: Yes

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

File: bindings/nodejs/lib/types/block/core/block-body.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { BasicBlockBody } from './basic';
+import { ValidationBlockBody } from './validation';
+/**
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/core/block-body.ts:7]:
```suggestion

/**
```

File: bindings/nodejs/lib/types/block/core/block.ts [hunk 3]
```diff
@@ -1,67 +1,206 @@
...
- * All of the block types.
+ * Represent the object that nodes gossip around the network.
  */
-export enum BlockType {
-    /// A Basic block.
-    Basic = 0,
-    /// A Validation block.
-    Validation = 1,
-}
+class Block {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/block/core/block.ts]: This is missing a proper `header` for the serialisation?

Comment (Thoralf-M) [reply bindings/nodejs/lib/types/block/core/block.ts]: Changed

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**